sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

```diff
@@ -259,25 +329,70 @@ private async Task<Response> SendRequestAsync(Request request, CancellationToken
                     throw await response.CreateRequestFailedExceptionAsync();
             }
         }
+        private Response SendRequest(Request request, CancellationToken cancellationToken)
+        {
+            var response = _pipeline.SendRequest(request, cancellationToken);
 
+            switch (response.Status)
+            {
+                case 200:
+                case 201:
+                    return response;
+                default:
+                    throw response.CreateRequestFailedException();
+            }
+        }
 
-        private async Task<Response<Page<T>>> GetPageAsync<T>(Uri pageUri, Func<T> itemFactory, CancellationToken cancellationToken)
-            where T : Model
+        private async Task<PageResponse<T>> GetPageAsync<T>(Uri firstPageUri, string nextLink, Func<T> itemFactory, CancellationToken cancellationToken)
+                where T : Model
         {
-            using (Request request = CreateRequest(HttpPipelineMethod.Get, pageUri, addAPIVersion:false))
+            // if we don't have a nextLink specified, use firstPageUri
+            if (nextLink != null)
+            {
+                firstPageUri = new Uri(nextLink);
+            }
+
+            using (Request request = CreateRequest(HttpPipelineMethod.Get, firstPageUri, addAPIVersion: false))
             {
                 Response response = await SendRequestAsync(request, cancellationToken);
 
-                return this.CreateResponse(response, new Page<T>(itemFactory));
+                // read the respose
+                Page<T> responseAsPage = new Page<T>(itemFactory);
+                responseAsPage.Deserialize(response.ContentStream);
+
+                // convert from the Page<T> to PageResponse<T>
+                return new PageResponse<T>(responseAsPage.Items.ToArray(), response, responseAsPage.NextLink?.ToString());
+            }
+        }
+
+        private PageResponse<T> GetPage<T>(Uri firstPageUri, string nextLink, Func<T> itemFactory, CancellationToken cancellationToken)
+            where T : Model
+        {
+            // if we don't have a nextLink specified, use firstPageUri
+            if (nextLink != null)
+            {
+                firstPageUri = new Uri(nextLink);
+            }
+
+            using (Request request = CreateRequest(HttpPipelineMethod.Get, firstPageUri, addAPIVersion: false))
+            {
+                Response response = SendRequest(request, cancellationToken);
+
+                // read the respose
+                Page<T> responseAsPage = new Page<T>(itemFactory);
+                responseAsPage.Deserialize(response.ContentStream);
+
+                // convert from the Page<T> to PageResponse<T>
```

Comment:
We need to get rid of `Page<T>` completely.

Reply:
Hm, this ties into the question of having a `Model` base type at all. Let's skip for this PR.

Reply:
Issue https://github.com/Azure/azure-sdk-for-net/issues/6477 created to address that change. I added both `Model` and `Page`